### PR TITLE
chore!: add shell linting to uds-common linting

### PR DESCRIPTION
**Pull request by Racer159**
This adds shellcheck for linting shell scripts in UDS Packages.

> [!CAUTION]
> :warning: **BREAKING CHANGE** - shellcheck is on by default and will now fail linting if a package has scripts that do not comply.

### CODE REVIEW

File: tasks/lint.yaml [hunk 1]
```diff
@@ -20,4 +27,93 @@ tasks:
     description: Run linting checks on OSCAL
     actions:
       - description: Lint OSCAL files
-        cmd: lula tools lint -f $OSCALFILES
+        cmd: lula tools lint -f "$OSCALFILES"
```
Comment (andrewg-xyz) [inline tasks/lint.yaml:30]: iirc this breaky if the '$OSCALFILES' don't exist

Comment (Racer159) [reply tasks/lint.yaml:30]: I can add an if on it - _and_ now that we have this https://github.com/defenseunicorns/uds-common/blob/main/docs/package_integration/oscal-guidelines.md I'll update the default too.

Comment (Racer159) [reply tasks/lint.yaml]: added but @brandtkeller thoughts on including this check in Lula directly? - also maybe not bc I think eventually we'd want to force this on you...

Comment (Racer159) [reply tasks/lint.yaml]: added a warning for now - we may eventually want to force this on people.

Comment (Racer159) [reply tasks/lint.yaml]: (also this is broken until https://github.com/defenseunicorns/uds-common/pull/254 pulls in an update to uds-cli)

**Comment by Racer159** [comment]
(scan lint is blocked until uds cli update comes through https://github.com/defenseunicorns/uds-common/pull/254)

**Review by andrewg-xyz — APPROVED** [review]

**Review by zachariahmiller — DISMISSED** [review]

**Review by zachariahmiller — APPROVED** [review]

**Pull request merged**